### DISCUSSION TITLE
Is ReactiveUI good for building native .NET iOS/Android apps with ReactiveUI (not MAUI, equivalent to Xamarin.iOS/Xamarin.Android)

### DISCUSSION THREAD

**Discussion by kyurkchyan** [reactiveui/ReactiveUI#3755] [Q&A]
Dear ReactiveUI community

Back in the old days, when there was no Xamarin Forms or MAUI, I developed Xamarin apps with Xamarin.iOS and Xamarin.Android native. Which meant, shared core/view models and native UIs. 

This approach had both pros and cons. The pro was that you can have a close-to-native performance. The con well you develop the UI twice. 

Currently, we've launched the production-ready app on the .NET MAUI, and we are not satisfied with the results. Collection views are especially awful. But generally, the UI rendering is pretty noticeably worse than native applications. 

Considering this, we want to move some of our UIs to .NET iOS/Android native. We are using ReactiveUI as our core library. Hence, a question arises: How well does ReactiveUI support native iOS/Android views? I am especially curious about the support of collection view on iOS and Recycler view on Android data, which should be bound to `ObservableCollection` or `DynamicData`. Are these libraries being maintained? Are they production grade? 

I could not find any recent nice examples of ReactiveUI usage in .NET native iOS/Android scenarios. Are there any good examples of how one can utilize ReactiveUI on .NET native iOS/Android with relatively good complexity to demonstrate the ReactiveUI capabilities?

**Comment by anaisbetts** [comment]
There is typically very little interest around working on these two platforms directly - most people use XamForms or Maui, but afaik the recycler view code still works. An important thing to know about ReactiveUI though, is that you are never ever actually bound by waiting for built-in support for anything, ReactiveUI is a lot of individual pieces, all replaceable at any time. So, if the binding to RecyclerView isn't working for you, copy-paste the file into your app and start making changes - it'll work just as well as the built-in one. 

There is very little "special sauce" that you can't create on your own in your app - you can literally add support for a completely new UI platform solely from an app, without touching the ReactiveUI source code!

**Reply by kyurkchyan** [reply]
Thanks, @anaisbetts, for the reply.

Indeed, that's why I love ReactiveUI. It's an outstanding balance between full-blown end-to-end and pure view-model frameworks. And I've used this customizability many times in my project. 

While extensibility is possible, everything has its cost. The reason I've raised this topic of discussion is to see if anybody has gone through utilizing ReactiveUI for the .NET native scenarios and share their experience. And also, it'd be cool to see a sample repo with ReactiveUI used in .NET native.
